extensions/wikia/AdEngine/js/spec/slot/resolvedState.spec.js

```diff
@@ -155,113 +117,138 @@ describe('ext.wikia.adEngine.slot.resolvedState', function () {
 		);
 	}
 
-	testCases.forEach(function (testCase) {
-		var testName = 'Should return ' + testCase.expected + ' when params: ' + JSON.stringify(testCase.params) +
-			' and resolvedState query param equals: ' + testCase.queryParam;
-
-		it(testName, function () {
+	blockingUrlParams.forEach(function (params) {
+		it('should not be in resolved state when is not blocked by query param ' + params, function () {
 			spyOn(mocks.qs, 'getVal');
+
+			mocks.qs.getVal.and.returnValue(params);
+
 			var rs = getModule();
-			mocks.qs.getVal.and.returnValue(testCase.queryParam);
 
-			expect(testCase.expected).toEqual(rs.setImage(testCase.params).backgroundImage.src);
+			expect(rs.isResolvedState()).toEqual(false);
 		});
 	});
 
-	testCasesWithTwoAssets.forEach(function (testCase) {
-		var testName = 'Should return ' + testCase.expectedImage1 + ' and ' + testCase.expectedImage2 +
-			' when params: ' + JSON.stringify(testCase.params) + ' and resolvedState query param equals: ' +
-			testCase.queryParam;
-
-		it(testName, function () {
+	forcingUrlParams.forEach(function (params) {
+		it('should be in resolved state when is forced by query param ' + params, function () {
 			spyOn(mocks.qs, 'getVal');
+
+			mocks.qs.getVal.and.returnValue(params);
+
 			var rs = getModule();
-			mocks.qs.getVal.and.returnValue(testCase.queryParam);
 
-			expect(testCase.expectedImage1).toEqual(rs.setImage(testCase.params).image1.element.src);
-			expect(testCase.expectedImage2).toEqual(rs.setImage(testCase.params).image2.element.src);
+			expect(rs.isResolvedState()).toEqual(true);
 		});
 	});
 
-	it('Should update params aspect ratio', function () {
+	it('should not be in resolved state when no information about seen ad was stored', function () {
+		spyOn(mocks.cache, 'get');
+		spyOn(mocks.cache, 'set');
+
+		mocks.cache.get.and.returnValue(null);
+
 		var rs = getModule();
 
-		expect(rs.setImage(createCorrectParams()).aspectRatio).toEqual(RESOLVED_STATE_ASPECT_RATIO);
+		expect(rs.isResolvedState()).toEqual(false);
 	});
 
-	it('Should update image src', function () {
-		var params = createCorrectParams(),
-			rs = getModule();
+	it('should be in resolved state when information about seen ad was stored', function () {
+		spyOn(mocks.cache, 'get');
+		spyOn(mocks.cache, 'set');
+
+		mocks.cache.get.and.returnValue({
+			lastSeenDate: new Date()
+		});
+
 
-		expect(rs.setImage(params).backgroundImage.src).toEqual(params.resolvedState.imageSrc);
+		var rs = getModule();
+
+		expect(rs.isResolvedState()).toEqual(true);
 	});
 
-	it('Should not update image if there is no background image (template without backgroundImage)', function () {
-		var params = {},
+	it('should not modify params if template does not support resolved state', function() {
+		spyOn(mocks.videoSettings, 'getParams');
+
+		var params = createIncorrectParams(),
 			rs = getModule();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove the declaration of the unused 'rs' variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedVariable)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "rs" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS1854)